### ISSUE TITLE
Remove the deprecated twisted.web.http.Request.headers/request_headers

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8136]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8136|
|Type|enhancement|
|Created|2015-12-04 08:04:41Z|
|Branch|https://github.com/twisted/twisted/tree/oldheaders-remove-py3-8136|

They've been deprecated since 13.2, and are a bad interface.

<details><summary>Searchable metadata</summary>

```
trac-id__8136 8136
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__Twisted_16_0 Twisted 16.0
branch__branches_oldheaders_remove_py3_8136 branches/oldheaders-remove-py3-8136
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__web web
keywords__None None
time__1449216281694127 1449216281694127
changetime__1449381853566932 1449381853566932
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [46404]) Branching to oldheaders-remove-py3-8136.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

Builds look good. Please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Looks good. Thanks!

Please merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl commented|
|-|-|

Milestone renamed

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46437]) Merge oldheaders-remove-py3-8136: Remove the deprecated headers and received_headers attributes from t.w.h.Request.

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8136](https://github.com/twisted/twisted/issues/8136)
